### ISSUE TITLE
Doesn't sync from Nextcloud WebDav

### ISSUE THREAD

**Issue by g-vdn** [laurent22/joplin#6427]
## Environment

Joplin app installed from Play Store
Phone: Huawei Mate 20 Pro running EMUI 12

## Steps to reproduce

1. Installed Joplin, created some notes 
2. Created sync with Nextcloud, tested if all is in order
3. Uninstalled and reinstalled the app on phone, redone the sync steps. Freshly created notes gone 

## Describe what you expected to happen

Notes to be available after reinstall and sync

**Comment by Daeraxa** [comment]
NextCloud definitely works just fine with Joplin in general but there have been issues with certain providers before.
Are the notes still in your nextcloud? Anything suspicious in the logs?

**Comment by g-vdn** [comment]
Nothing out of the ordinary in the server logs, I host my own Nextcloud via docker / unraid. I'll try to tinker with it more tomorrow. Behavior is strange, the app says that sync is complete, I can see the folder in nextcloud increasing in size, I uninstall the android app, install again, sync via nextcloud webdav, and nothing...

**Comment by Daeraxa** [comment]
You are definitely leaving the android app open whilst it syncs? Normally something thats an issue when you have loads of data rather than a tiny bit because it cannot background sync. 
I have my own Nextcloud hosted hosted but don't have any issues.

**Comment by g-vdn** [comment]
I don't have loads of data, I create for test purpose just 2-3 notes and a new notebook. After that sync, and when message says synced, i uninstall and reinstall the app on phone. After reinstalling and entering the webdav again, I sync, and only the default notes appear, nothing which I created previously. No errors appear whatsoever.

**Comment by Daeraxa** [comment]
Are the .md files etc. still present on the sync target after the failed sync?
Can you provide logs?

**Comment by g-vdn** [comment]
Ok, so I have no idea why it's working like this, but it does. Changing from Nextcloud to WebDav in the sync options did the trick. I can see the notes created from PC and phone after sync. I used same WebDav URL, user, pass with Nextcloud but that didn't work for me. I'm not 100% sure, but I have a hunch it's my phone, had some troubles lately with other apps, and with the help of reditors, I reached this conclusion. I replicated all steps in my girlfriends phone with Nextcloud set as sync option, and it is working just fine. Wanna buy a phone? I'm changing mine :)

**Comment by Daeraxa** [comment]
How odd. I'm also not 100% how well using the webDAV sync target with NextCloud will work because NextCloud has its own... unique... implementation of WebDAV (I think its lock method is different).

**Comment by g-vdn** [comment]
I'm happy that finally worked,and hope will not have same issue with other phone. Thanks for your help and time

**Comment by dpoulton-github** [comment]
@Daeraxa 

Sorry for being late to the party on this but using the WEBDAV connector and not the Nextcloud one has been something I have suggested on the forum for a couple of years and people have marked it as a solution. This is a more recent post:

https://discourse.joplinapp.org/t/joplin-app-nextcloud-connecting-not-possible/16281

I had been syncing using the Nextcloud connector for some time but around the time of the release of NC18 something possibly changed in NC. I believe that it was around the time NC changed the format of their webdav urls. Anyway I had started noticing some strange behaviours. From then on I used the webdav connector on all devices (Windows, Linux, Google Android, Lineage Android) with no problems. I don't have the knowledge to know why it worked better, I just tried it out and it did.
